scripts/nbconverted/3_add_batch_effects.py

```diff
@@ -0,0 +1,214 @@
+
+            # Add batch effect
+            subset_genes_to_change_tile = pd.DataFrame(
+                pd.np.tile(
+                    subset_genes_to_change,
+                    (num_samples_per_batch, 1)),
+                index=batch_df.index)
+
+            offset_vector = pd.DataFrame(subset_genes_to_change_tile*stretch_factor)
+            offset_vector.columns = offset_vector.columns.astype(str)
+            batch_df = batch_df + offset_vector
+
+            # if any exceed 1 then set to 1 since gene expression is normalized
+            batch_df[batch_df>=1.0] = 1.0
+
+
+            # Append batched together
+            batch_data_df = batch_data_df.append(batch_df)
+
+            # Select a new direction (i.e. a new subset of genes to change)
+            np.random.shuffle(subset_genes_to_change)
+            
+        # Save
+        batch_data_df.to_csv(batch_file, sep='\t')
+
+
+# ## Plot batch data using UMAP
+
+# In[7]:
+
+
+# Plot generated data 
+
+for i in num_batches:
+    batch_data_file = os.path.join(
+        os.path.dirname(os.getcwd()),
+        "data",
+        "batch_simulated",
+        analysis_name,
+        "Batch_"+str(i)+".txt")
+    
+    batch_data = pd.read_table(
+        batch_data_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+    
+    # UMAP embedding of decoded batch data
+    batch_data_UMAPencoded = umap.UMAP().fit_transform(batch_data)
+    batch_data_UMAPencoded_df = pd.DataFrame(data=batch_data_UMAPencoded,
+                                             index=batch_data.index,
+                                             columns=['1','2'])
+    
+        
+    g = ggplot(aes(x='1',y='2'), data=batch_data_UMAPencoded_df) +                 geom_point(alpha=0.5) +                 scale_color_brewer(type='qual', palette='Set2') +                 ggtitle("{} Batches".format(i))
+    
+    print(g)
+
+
+# ## Plot batch data using PCA
+
+# In[8]:
+
+
+# Plot generated data 
+
+for i in num_batches:
+    batch_data_file = os.path.join(
+        os.path.dirname(os.getcwd()),
+        "data",
+        "batch_simulated",
+        analysis_name,
+        "Batch_"+str(i)+".txt")
+    
+    batch_data = pd.read_table(
+        batch_data_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    # PCA projection
+    num_PCs = 2
+    pca = PCA(n_components=num_PCs)
+
+    # Use trained model to encode expression data into SAME latent space
+    batch_data_PCAencoded = pca.fit_transform(batch_data)
+    
+    
+    batch_data_PCAencoded_df = pd.DataFrame(batch_data_PCAencoded,
+                                         index=batch_data.index,
+                                         columns=['1', '2'])
+    
+    g = ggplot(aes(x='1',y='2'), data=batch_data_PCAencoded_df)  +                 geom_point(alpha=0.5) +                 scale_color_brewer(type='qual', palette='Set2') +                 ggtitle("{} Batches".format(i))
```

Comment:
looks good considering we are projecting only the two largest variance linear dimensions